src/core/mesh/qgstriangularmesh.h

```diff
@@ -118,6 +120,40 @@ class CORE_EXPORT QgsTriangularMesh
 
     QVector<QVector3D> vertexNormals( float vertScale ) const;
 
+    /**
+     * Returns simplify meshes. The first simplified mesh is simplified with a goal of a number of triangles equals to the
+     * number of triangles of the base mesh divised by the reduction factor. For the following mesh the same reduction factor is used with
+     * the prededent goal of number of triangles.
+     * There are as much simplified meshes as necessary to have a the minimum triangles count on the last simplified mesh
+     *
+     * \param reductionFactor is the factor used to reduce the number of triangles of the mesh
+     * \param minimumTrianglesCount is the minimal faces count on simplified mesh
+     * \since QGIS 3.14
+     */
+    QVector<QgsTriangularMesh *> simplifyMesh( int reductionFactor, int minimumTrianglesCount = 10 ) const;
+
+    /**
+     * Returns the average size of triangles in map unit. It is calculated using the maximum of the height/width of the
+     * bounding box of each triangles.
+     *
+     * \since QGIS 3.14
+     */
+    double averageTriangleSize() const;
+
+    /**
+     * Returns the corresponding index of level of details on which this mesh is associated
+     *
+     * \since QGIS 3.14
+     */
+    int lod() const;
```

Comment:
better to rename to levelOfDetail()

Can we add more detail about these values? e.g. lod=0 means the first level of simplification and lod=1 is the mesh after two rounds of simplificiation?